rust/src/milla/api.rs

```diff
@@ -434,19 +451,71 @@ pub(crate) fn internal_reset_superconductivity(x: i32, y: i32, z: i32) -> Result
     Ok(())
 }
 
+/// BYOND API for a heat source creating a hotspot on a tile.
+#[byondapi::bind]
+fn milla_create_hotspot(turf: ByondValue, temperature: ByondValue, volume: ByondValue) {
+    let (x, y, z) = byond_xyz(&turf)?.coordinates();
+    let rust_temperature = conversion::bounded_byond_to_option_f32(temperature, 0.0, f32::INFINITY)?.ok_or(eyre!("Hotspot temperature is required.."))?;
+    let rust_volume = conversion::bounded_byond_to_option_f32(volume, 0.0, f32::INFINITY)?.ok_or(eyre!("Hotspot volume is required.."))?;
+
+    internal_create_hotspot(x as i32 - 1, y as i32 - 1, z as i32 - 1, rust_temperature, rust_volume / TILE_VOLUME)?;
+    Ok(Default::default())
+}
+
+/// Rust version of a heat source creating a hotspot.
+pub(crate) fn internal_create_hotspot(x: i32, y: i32, z: i32, temperature: f32, volume: f32) -> Result<()> {
+    let buffers = BUFFERS.get().ok_or(eyre!("BUFFERS not initialized."))?;
+    let active = buffers.get_active().read().unwrap();
```

Comment:
Is there a reasonable chance that this might not be Ok/Some? Feels like it might be worth doing another `ok_or()` check on this to avoid the off-chance of a panic that tanks the whole atmos system.

Reply:
We never acquire a write lock on either buffer, so .read() can never fail. The only reason we're using a lock here is so that get_active and get_inactive can return a reference to their respective buffers without scaring Rust (otherwise, it looks fishy concurrency-wise).

OnceLock would work, too, but it'd just change .read() to .get(), so I'm not really fussed about changing it.

Reply:
Okay, correction: We do call write() during initialization of a Z level, which could technically happen after the system starts. However, in that case, we would block until it finished. The only way we would panic here is:
> This function will return an error if the RwLock is poisoned. An RwLock is poisoned whenever a writer panics while holding an exclusive lock. The failure will occur immediately after the lock has been acquired.
> From: https://doc.rust-lang.org/std/sync/struct.RwLock.html#method.read

So this will only panic if we panic during write, and our only writer can't panic.

Reply:
(Plus, the writer is also a BYOND call, so it shouldn't be possible to end up here while it's writing.)